bedrock/base/middleware.py

```diff
@@ -10,57 +10,174 @@
+            return self._redirect(request, cleaned_lang_via_querystring, subpath)
+
+        if not lang_code and path_needs_lang_code(request.path) and not is_root_path_with_no_language_clues(request):
+            lang_code = get_language_from_headers(request)
+            return self._redirect(request, lang_code, subpath)
+
+        if lang_code and lang_code_changed:
+            return self._redirect(request, lang_code, subpath)
+
+        request.locale = lang_code if lang_code else ""
+
+
+class BedrockLocaleMiddleware(DjangoLocaleMiddleware):
+    """Middleware that usually* wraps Django's own i18n middleware in order to
+    ensure we normalize language codes - i..e. we ensure they are in
+    mixed case we use, rather than Django's internal all-lowercase codes.
+
+    *for one specific situation, though, we skip Django's LocaleMiddleware entirely:
+    we have a special SEO-helping page that gets returned to robots/spiders that
+    don't declare an accept-language header, showing a list of locales to pick.
+
+    It needs to be kept super-light so that it doesn't diverge too far from
+    the stock LocaleMiddleware, lest we wake up dragons when we use
+    wagtail-localize, which squarely depends on django's LocaleMiddleware.
+
+    Note: this is not SUMO's LocaleMiddleware, this just a tribute.
+    (https://github.com/escattone/kitsune/blob/main/kitsune/sumo/middleware.py#L128)
     """
 
-    def __init__(self, get_response=None):
-        if not settings.USE_L10N:
-            warn(
-                """
-                The `USE_L10N` setting is False but LocaleURLMiddleware is
-                loaded. Consider removing bedrock.base.middleware.LocaleURLMiddleware
-                from your MIDDLEWARE setting.
-                """.strip()
+    def process_request(self, request):
+        if is_root_path_with_no_language_clues(request):
+            # Skip using Django's LocaleMiddleware
+            metrics.incr(
+                "bedrock.localemiddleware.skipdjangolocalemiddleware",
+                tags=[f"path:{request.path}"],
             )
-        self.get_response = get_response
+        else:
+            with normalized_get_language():
+                with simplified_check_for_language():
+                    return super().process_request(request)
 
-    def __call__(self, request):
-        response = self.process_request(request)
-        if response:
+    def process_response(self, request, response):
+        if is_root_path_with_no_language_clues(request):
+            # Skip using Django's LocaleMiddleware on the response cycle too
             return response
-        return self.get_response(request)
 
-    def process_request(self, request):
-        prefixer = urlresolvers.Prefixer(request)
-        urlresolvers.set_url_prefix(prefixer)
+        with normalized_get_language():
+            with simplified_check_for_language():
+                return super().process_response(request, response)
+
```

Comment:
**praise:** Credit for the patcing approach goes to @escattone, who worked out how to do it for SUMO